src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/InternalCaseCreationContactDetailsAppender.java

```diff
@@ -26,7 +26,9 @@ public boolean canHandle(
         requireNonNull(callback, "callback must not be null");
 
         return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
-               && Arrays.asList(Event.START_APPEAL, Event.EDIT_APPEAL).contains(callback.getEvent());
+               && Arrays.asList(Event.START_APPEAL,
+                Event.EDIT_APPEAL,
+                Event.EDIT_APPEAL_AFTER_SUBMIT).contains(callback.getEvent());
```

Comment:
Why is the new event added here - EDIT_APPEAL_AFTER_SUBMIT? Can we update the APPEAL_SUBMISSION_INTERNAL_DATE or TRIBUNAL_RECEIVED_INTERNAL_DATE or HOME_OFFICE_DECISION_DATE  after submission?
Also why don't we use HOME_OFFICE_DECISION_DATE directly? Is there a conflict of the label?

Reply:
We can edit the values for APPEAL_SUBMISSION_INTERNAL_DATE and TRIBUNAL_RECEIVED_INTERNAL_DATE and HOME_OFFICE_DECISION_DATE in the EDIT_APPEAL_AFTER_SUBMIT event. 
Correct - we are not using the HOME_OFFICE_DECISION_DATE field because it requires the label to be "Appeal created by the Tribunal" for  isAdmin AND (ADA OR Detained(non ADA, non AAA)) else the label should say  'Appeal submitted'.